src/fixes/unordered_key.rs

```diff
@@ -0,0 +1,724 @@
+use super::Fix;
+use crate::common::*;
+
+pub(crate) struct UnorderedKeyFixer<'a> {
+    name: &'a str,
+}
+
+impl Default for UnorderedKeyFixer<'_> {
+    fn default() -> Self {
+        Self {
+            name: "UnorderedKey",
+        }
+    }
+}
+
+// When we sort the keys, we handle a significant line (with key) with all previous comments
+// as a whole. E. g.:
+// ```
+// B=C
+// # Comment
+// A=B
+// ```
+// will be fixed to:
+// ```
+// # Comment
+// A=B
+// B=C
+// ```
+// We check the order separately in each group of entries (groups are separated by blank lines or
+// control comments).
+impl Fix for UnorderedKeyFixer<'_> {
+    fn name(&self) -> &str {
+        self.name
+    }
+
+    fn fix_warnings(
+        &mut self,
+        warnings: Vec<&mut Warning>,
+        lines: &mut Vec<LineEntry>,
+    ) -> Option<usize> {
+        // We find all sorting groups and sort them
+
+        let mut start_index = 0;
+        let mut end = None;
+        let mut is_disabled = false;
+
+        for i in 0..lines.len() {
+            let line = &lines[i];
+
+            let mut is_control_comment = false;
+            let mut controls_this_check = false;
+            let mut is_off = false;
+
+            if let Some(comment) = line.get_control_comment() {
+                is_control_comment = true;
+                controls_this_check = comment.checks.contains(&self.name);
+                is_off = comment.is_disabled();
+            }
+
+            if !is_disabled {
+                if !line.is_empty_or_comment() {
+                    end.replace(i + 1);
+                }
+
+                let is_last = i == lines.len() - 1;
+                if line.is_empty() || is_last || is_control_comment {
```

Comment:
```suggestion
                if line.is_empty() || line.is_last_line() || is_control_comment {
```
Why don't you use the `is_last_line` method?

Reply:
Now we don't delete lines (just mark them), so this is OK.

Reply:
Done